code/modules/shuttle/shuttle.dm

```diff
@@ -704,6 +748,7 @@
 			if(prearrivalTime && mode != SHUTTLE_PREARRIVAL)
 				set_mode(SHUTTLE_PREARRIVAL)
```

Comment:
This would be where you could add a proc call on the docking_port for a new OnStartPrearrival or something.